music21/midi/translate.py

```diff
@@ -349,28 +411,23 @@ def midiEventsToNote(eventList, ticksPerQuarter=None, inputM21=None) -> note.Not
     else:
         raise TranslateException(f'cannot handle MIDI event list in the form: {eventList!r}')
 
-    # here we are handling an issue that might arise with double-stemmed notes
-    if (tOff - tOn) != 0:
-        if inputM21 is None:
-            n = note.Note(duration=ticksToDuration(tOff - tOn, ticksPerQuarter))
-        else:
-            n = inputM21
-            n.duration = ticksToDuration(tOff - tOn, ticksPerQuarter, n.duration)
-    else:
-        # environLocal.printDebug(['cannot translate found midi event with zero duration:', eOn, n])
-        # for now, substitute grace note
-        if inputM21 is None:
-            n = note.Note()
-        else:
-            n = inputM21
-        n.getGrace(inPlace=True)
-
-    n.pitch.midi = eOn.pitch
-    n.volume.velocity = eOn.velocity
-    n.volume.velocityIsRelative = False  # not relative coming from MIDI
+    nr = _constructOrUpdateNotRest(
+        eOn, tOn, tOff, ticksPerQuarter, inputM21=inputM21, preferredClass=note.Note)
+
+    if isinstance(nr, note.Note):
+        nr.pitch.midi = eOn.pitch
+    elif isinstance(nr, note.Unpitched):
+        display_pitch = pitch.Pitch(eOn.pitch)
+        nr.displayStep = display_pitch.step
+        nr.displayOctave = display_pitch.octave
```

Comment:
this does not seem correct -- we would rarely want to display a percussion note at MIDI pitch.  I think that somewhere in music21 there's a mapper from instruments to positions on staves.

Reply:
Thanks, looks like it's the `.percMapPitch` of an `UnpitchedPercussion` instance. So there could be a cleanup routine that could live elsewhere that MIDI parsing calls at the end to set displayPitch from the percMapPitch of the closest instrument. I'll track on another issue.

Reply:
Never mind, I think I can do it quickly on this branch.

Reply:
Actually, no I can't find staff positions anywhere. `percMapPitch` _isn't_ it.